docs/user/syncing-apprepository-webhook.md

```diff
@@ -0,0 +1,159 @@
+A number of platforms do use webhooks for triggering actions when something occurs. For instance, [Harbor notifies the webhook endpoint of certain events that occur in the project]
+(https://goharbor.io/docs/1.10/working-with-projects/project-configuration/configure-webhooks/).
+Webhook notifications provide information about events in JSON format and are usually delivered by HTTP(s) POST to an existing webhook endpoint URL.
+
+This document will use Harbor as a running example for explaining how a webhook is configured for triggering an App Repository syncing process.
+> In other platforms the process will be pretty similar. Doubts? Please feel free to [open an issue](https://github.com/kubeapps/kubeapps/issues/new) if you need further guidance!
+
+
+## TL; DR
+
+- Create the ServiceAccount, grant with sufficient permissions and get the token:
+```bash
+cat <<EOF | kubectl apply -f -
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: apprepositories-webhook
+---
+kind: ClusterRole
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: apprepositories-webhook
+rules:
+  - apiGroups:
+      - kubeapps.com
+    resources:
+      - apprepositories
+    verbs:
+      - get
+      - update
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  name: apprepositories-webhook
+  namespace: kubeapps
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: apprepositories-webhook
+subjects:
+- kind: ServiceAccount
+  name: apprepositories-webhook
+  namespace: default
+EOF
+```
+
+- Get the `<TOKEN>`:
+```bash
+kubectl get secret $(kubectl get serviceaccount apprepositories-webhook -o jsonpath='{range .secrets[*]}{.name}{"\n"}{end}' | grep apprepositories-webhook) -o jsonpath='{.data.token}' -o go-template='{{.data.token | base64decode}}' && echo
+```
+
+- In your platform:
+	- Webhook URL:    `https://<KUBEAPPS_URL>/api/v1/clusters/<CLUSTER_NAME>/namespaces/<NAMESPACE_NAME>/apprepositories/<APPREPO_NAME>/refresh`
+	- Method: `POST`
+	- Authentication header: `Bearer <TOKEN>`
+
+> For instance: `https://<KUBEAPPS_URL>/api/v1/clusters/default/namespaces/kubeapps/apprepositories/bitnami/refresh` will update the App Repository `bitnami` in the namespace `kubeapps` of the cluster `default`. 
+
+
+
+## Creating and granting a ServiceAccount for authenticating requests
+
+Since the webhook endpoint will be used outside the user browser session (i.e., no OICD login), an ad-hoc ServiceAccount ought to be created; next, it has to be granted with sufficient permissions to `get` and `update` the `AppRepository` objects. Finally, the ServiceAccount token must be retrieved.
+
+### Creating the Kubernetes ServiceAccount, ClusterRole and RoleBinding objects
+
+First, the `ServiceAccount` and the `ClusterRole` objects are created. In this example, the ServiceAccount *apprepositories-webhook* is being created.
+
+```bash
+cat <<EOF | kubectl apply -f -
```

Comment:
Is there a reason that you're not explicitly putting the service account in a specific namespace of the AppRepository? (I mean, I know it can go in any namespace, but I thought it'd be clearer to keep them together in the one).

Reply:
I just intended to express that the actual ServiceAccount is independent of whichever chosen namespace for the RoleBinding and one is able to grant this sa to any namespace.
I'm not sure here whether it'd be clearer to use the same namespace; don't know :S